### PR TITLE
Add support for 64-bit numbers and size_t to wxConfig

### PR DESCRIPTION
**Pull request by vadz**
See [#19091](https://trac.wxwidgets.org/ticket/19091) but beyond this lack of support for 64-bit numbers is a bit ridiculuous in 2021.

### CODE REVIEW

File: interface/wx/config.h [hunk 1]
```diff
@@ -549,6 +549,51 @@ class wxConfigBase : public wxObject
     */
     bool Read(const wxString& key, long* l,
               long defaultVal) const;
+    /**
+        Reads a 64-bit long long value, returning @true if the value was found.
```
Comment (PBfordev) [inline interface/wx/config.h:553]: Perhaps everywhere where there is something like "value was/was not found", "value" should be replaced with "key"? 

I think "key" is already used like this in wxConfigBase docs and using "value" can bit a bit confusing when we also have an argument named "value" as well.

File: interface/wx/config.h [hunk 2]
```diff
@@ -549,6 +549,51 @@ class wxConfigBase : public wxObject
...
+        If the value was not found, @a defaultVal is used instead.
+
+        @since 3.1.5
+
+        @beginWxPerlOnly
+        Not supported by wxPerl.
+        @endWxPerlOnly
+    */
+    bool Read(const wxString& key, wxLongLong_t* ll,
```
Comment (PBfordev) [inline interface/wx/config.h:573]: The function declaration seems to be incomplete, should probably look like
```suggestion
    bool Read(const wxString& key, wxLongLong_t* ll, wxLongLong_t defaultVal) const;
```

File: src/common/config.cpp [hunk 3]
```diff
@@ -174,6 +177,40 @@ bool wxConfigBase::Read(const wxString& key, int *pi, int defVal) const
     return r;
 }
 
+// size_t is stored either as long or long long (Win64)
+#if SIZEOF_SIZE_T == SIZEOF_LONG
```
Comment (PBfordev) [inline src/common/config.cpp:181]: Sorry for nitpicking, but as `size_t` is unsigned, should we not used `unsigned long` and `wxULongLong` or `wxUint64` here?

But perhaps the compilers will not complain...

File: interface/wx/config.h (hunk 2 above)
Comment (vadz) [reply interface/wx/config.h:573]: Oops, thanks for noticing!

File: src/common/config.cpp (hunk 3 above)
Comment (vadz) [reply src/common/config.cpp:181]: Handling of unsigned types in `wxConfig` is suspicious generally speaking, but I didn't want to change it now, so I've just done the strict minimum needed to allow using `size_t` with it at all.

**Comment by vadz** [comment]
Note: clang tests failure is unrelated.

File: interface/wx/config.h (hunk 1 above)
Comment (PBfordev) [reply interface/wx/config.h:553]: I see, I just looked at the first couple of getters in the docs, they used "key".  But they were for `wxString`s, the ones for numbers do you use "value".

**Pull request merged**